### PR TITLE
itest: adjust parallelization, poll intervals and timeouts to not starve goroutines

### PR DESCRIPTION
**Pull request by guggero**
Attempts to fix the `unable to send coins to Carol: unconfirmed utxo was not found in ListUnspent` error.

Background: Because of the introduced parallelization in the integration tests, it seems like some goroutines don't get enough CPU time anymore to do their jobs and we run into timeouts. This is accentuated by our very short poll intervals (20 milliseconds) for waits which spawns extra goroutines for each RPC request that is polled.
It also seems like `bitcoind` is more likely to run into this because it uses more goroutine for its notification listeners (at least I assume that's the case).

With this PR we tweak the parallelization to only run 4 parts at the same time and increases the poll interval by 10x to 200 milliseconds. Finally we also replace some of the hard coded timeouts to make it easy for darwin to increase them to pass the parallel tests as well locally.

### CODE REVIEW

File: lntest/itest/lnd_channel_backup_test.go [hunk 1]
```diff
@@ -987,6 +987,30 @@ func testChanRestoreScenario(t *harnessTest, net *lntest.NetworkHarness,
...
+		}
+
+		_, err = resp.Recv()
+		if err != nil && strings.Contains(err.Error(), expectedErr) {
+			return true
+		}
+
+		return false
+	}, defaultTimeout)
+	if err != nil {
```
Comment (cfromknecht) [inline lntest/itest/lnd_channel_backup_test.go:1010]: use `require`?

File: lntest/itest/lnd_channel_backup_test.go [hunk 2]
```diff
@@ -987,6 +987,30 @@ func testChanRestoreScenario(t *harnessTest, net *lntest.NetworkHarness,
...
+	// to find out if Dave is fully started.
+	err = wait.Predicate(func() bool {
+		const expectedErr = "must specify channel point"
+		resp, err := dave.CloseChannel(
+			ctxt, &lnrpc.CloseChannelRequest{},
+		)
+		if err != nil {
+			return false
+		}
+
```
Comment (cfromknecht) [inline lntest/itest/lnd_channel_backup_test.go:1002]: should there be a `defer resp.Close()` here?

**Review by cfromknecht — COMMENTED** [review]
💚💚💚

File: lntest/itest/lnd_channel_backup_test.go (hunk 2 above)
Comment (bhandras) [reply lntest/itest/lnd_channel_backup_test.go:1002]: I don't think client stream has a `Close`.

**Review by bhandras — APPROVED** [review]
LGTM!!

File: lntest/itest/lnd_channel_backup_test.go (hunk 2 above)
Comment (guggero) [reply lntest/itest/lnd_channel_backup_test.go:1005]: Added a `resp.CloseSend()` that's all we can do from the client side (and also cancel the context).

Comment (cfromknecht) [reply lntest/itest/lnd_channel_backup_test.go:1005]: had to look this up, there are apparently 4 ways to properly close a stream https://godoc.org/google.golang.org/grpc#ClientConn.NewStream. seems like cancelling the context is actually what we want here

Comment (cfromknecht) [reply lntest/itest/lnd_channel_backup_test.go:1005]: but i don't think `CloseSend` hurts anything sooo 🤷‍♂️

**Review by cfromknecht — APPROVED** [review]

**Pull request merged**